app/models/user.rb

```diff
@@ -44,12 +46,63 @@ def update_role(role_id)
     assignment.save!
   end
 
+  ###
+  # @description: Creates a token to rset the password of this user
+  ###
+  def generate_password_token!
+    update_columns(
+      reset_password_token: generate_token,
+      reset_password_sent_at: Time.now.utc
+    )
+  end
+
+  ###
+  # @description: Notify the user on how to reset the password
+  ###
+  def send_reset_password_instructions
+    UserMailer.with(user: self).forgot_pass.deliver_now
+  end
+
+  ###
+  # @description: Determines the validity of this user's password token
+  ###
+  def password_token_valid?
+    (reset_password_sent_at + 4.hours) > Time.now.utc
+  end
+
+  ###
+  # @description: Welcomes the user with an email which let's him sign in with
+  #   a new password.
+  ###
+  def send_welcome
+    UserMailer.with(user: User.last).welcome.deliver_now
```

Comment:
Why does it send a welcome to the last user? This may result in a race condition.